File: Pinz/Modules/PinzDomain/Sources/DesiredPlace.swift
import Foundation

public struct DesiredPlace: Hashable, Identifiable {
    public var id: String
    public var name: String
    public var description: String
    public var imageUrl: String?
    public let createdAt: Int

    public init(id: String, name: String, description: String, imageUrl: String? = nil, createdAt: Int = Int(Date().timeIntervalSince1970)) {
        self.id = id
        self.name = name
        self.description = description
        self.imageUrl = imageUrl
        self.createdAt = createdAt
    }
}

File: Backend/auth-service/internal/models/desired_place.go
package models

import "time"

// DesiredPlace — желаемое место пользователя. ImageURL хранит s3-ключ
// (пустой = картинка не загружена), наружу отдаётся presigned URL.
type DesiredPlace struct {
	ID          string
	UserID      string
	Name        string
	Description string
	ImageURL    string
	CreatedAt   time.Time
}
